src/main/webapp/linkWizard/SelectSampleConsent.js

```diff
@@ -109,6 +114,71 @@ export const SelectSampleConsent = hh(class SelectSampleConsent extends Componen
     this.setState({ documentOptions: documentOptions });
   };
 
+  getConsentGroups = () => {
+    this.setState({ consentGroupIsLoading: true });
+    ConsentGroup.getConsentGroupNames(this.props.getConsentGroups).then(
+      resp => {
+        const existingConsentGroups = resp.data.map(item => {
+          return {
+            key: item.id,
+            value: item.label,
+            label: item.label
+          }
+        });
+
+        this.setState({
+          existingConsentGroups: existingConsentGroups,
+          consentGroupIsLoading: false,
+          consentGroup: existingConsentGroups[0]
+        }, () => this.props.updateForm(this.state.consentGroup, "consentGroup"));
+        this.getAllSampleCollections(existingConsentGroups[0].key);
+      }
+    );
+  };
+
+  getAllSampleCollections = (consentKey) => {
+    this.setState({ sampleCollectionIsLoading: true });
+
+    SampleCollections.getCollectionsCGLinked(this.props.getConsentGroupSampleCollections, consentKey).then(
+      resp => {
+        const sampleCollectionList = [];
+        sampleCollectionList.push({label: "Sample Collections Linked to " + consentKey, options: []});
```

Comment:
This is confusing - you define a `const` and then immediately mutate it. Should it be a `let` instead?

Reply:
I think not. It's a convention to use every variable as constant if possible. In this case I do it this way because of this:

> Constants are not a restriction on the value itself, but on the variable's assignment of that value. In other words, the value is not frozen or immutable because of const, just the assignment of it.

[Source](https://github.com/getify/You-Dont-Know-JS/blob/master/es6%20%26%20beyond/ch2.md)

Reply:
Holy hell, I didn't realize that. I absolutely loathe this fact. It's even worse than what you quoted ... 
> The a variable doesn't actually hold a constant array; rather, it holds a constant reference to the array. The array itself is freely mutable.

Reply:
Even still, the author there recommends that you do not change `const` values once declared ... and he uses the same word I did, `confusing`
> My advice: to avoid potentially confusing code, only use const for variables that you're intentionally and obviously signaling will not change. In other words, don't rely on const for code behavior, but instead use it as a tool for signaling intent, when intent can be signaled clearly.

Reply:
Got it. No problem.